neqo-http3/tests/send_message.rs

```diff
@@ -0,0 +1,291 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use neqo_common::event::Provider;
+use neqo_crypto::AuthenticationStatus;
+use neqo_http3::{
+    ClientRequestStream, Error, Header, Http3Client, Http3ClientEvent, Http3Server,
+    Http3ServerEvent, Priority, Res,
+};
+use test_fixture::*;
+
+const RESPONSE_DATA: &[u8] = &[0x61, 0x62, 0x63];
+
+fn exchange_packets(client: &mut Http3Client, server: &mut Http3Server) {
+    let mut out = None;
+    loop {
+        out = client.process(out, now()).dgram();
+        out = server.process(out, now()).dgram();
+        if out.is_none() {
+            break;
+        }
+    }
+}
+
+fn receive_request(server: &mut Http3Server) -> Option<ClientRequestStream> {
+    while let Some(event) = server.next_event() {
+        if let Http3ServerEvent::Headers {
+            request,
+            headers,
+            fin,
+        } = event
+        {
+            assert_eq!(
+                headers.as_ref(),
+                &[
+                    Header::new(":method", "GET"),
+                    Header::new(":scheme", "https"),
+                    Header::new(":authority", "something.com"),
+                    Header::new(":path", "/")
+                ]
+            );
+            assert!(fin);
+            return Some(request);
+        }
+    }
+    None
+}
+
+fn send_trailers(request: &mut ClientRequestStream) -> Res<()> {
+    request.send_headers(&[
+        Header::new("something1", "something"),
+        Header::new("something2", "3"),
+    ])
+}
+
+fn send_headers(request: &mut ClientRequestStream) -> Res<()> {
+    request.send_headers(&[
+        Header::new(":status", "200"),
+        Header::new("content-length", "3"),
```

Comment:
See below: this should probably be a constant somehow.  Maybe you need to use lazy_static!() because I don't think that you can make a new Vec (as this does) without that.